Model/Service/Sync/AsyncBulkPublisher.php

```diff
@@ -0,0 +1,206 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Model\Service\Sync;
+
+use Exception;
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Framework\DataObject\IdentityGeneratorInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Serialize\SerializerInterface;
+use Magento\Store\Model\Store;
+use Nosto\Tagging\Model\ResourceModel\Product\Index\Collection as NostoIndexCollection;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Module\Manager;
+use Nosto\Tagging\Logger\Logger;
+
+// @codingStandardsIgnoreFile
+class AsyncBulkPublisher implements BulkSyncInterface
+{
+    const NOSTO_SYNC_MESSAGE_QUEUE = 'nosto_product_sync.update';
+    const BULK_SIZE = 100;
+    const STATUS_TYPE_OPEN = 4; //\Magento\Framework\Bulk\OperationInterface::STATUS_TYPE_OPEN;
+
+    /** @var \Magento\Framework\Bulk\BulkManagementInterface|null */
+    private $bulkManagement;
+
+    /** @var \Magento\AsynchronousOperations\Api\Data\OperationInterfaceFactory|null */
+    private $operationFactory;
+
+    /** @var IdentityGeneratorInterface */
+    private $identityService;
+
+    /** @var SerializerInterface */
+    private $serializer;
+
+    /** @var AsyncBulkConsumer */
+    private $asyncBulkConsumer;
+
+    /** @var Manager */
+    private $manager;
+
+    /**
+     * AsyncBulkPublisher constructor.
+     * @param IdentityGeneratorInterface $identityService
+     * @param SerializerInterface $serializer
+     * @param AsyncBulkConsumer $asyncBulkConsumer
+     * @param Manager $manager
+     * @param Logger $logger
+     */
+    public function __construct(
+        IdentityGeneratorInterface $identityService,
+        SerializerInterface $serializer,
+        AsyncBulkConsumer $asyncBulkConsumer,
+        Manager $manager,
+        Logger $logger
+    ) {
+        $this->identityService = $identityService;
+        $this->serializer = $serializer;
+        $this->asyncBulkConsumer = $asyncBulkConsumer;
+        $this->manager = $manager;
+        try {
+            $this->bulkManagement = ObjectManager::getInstance()
+                    ->get(\Magento\Framework\Bulk\BulkManagementInterface::class);
+            $this->operationFactory = ObjectManager::getInstance()
+                    ->get(\Magento\AsynchronousOperations\Api\Data\OperationInterfaceFactory::class);
+        } catch (Exception $e) {
+            $logger->info('Module Magento_AsynchronousOperations not available');
```

Comment:
Switch to debug perhaps? Otherwise we'll flood the logs on every product update.